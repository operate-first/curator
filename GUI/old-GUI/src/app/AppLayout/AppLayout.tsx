import * as React from 'react';
import { NavLink } from 'react-router-dom';
import {
  Nav,
  NavList,
  NavItem,
  Page,
  PageHeader,
  PageSidebar,
  SkipToContent,
  PageHeaderTools,
  Button
} from '@patternfly/react-core';
import { routes } from '@app/routes';
import { Role, RoleMap } from '@app/index';
import styles from '@patternfly/react-styles/css/components/Page/page';
import { css } from '@patternfly/react-styles';

interface IAppLayout {
  children: React.ReactNode;
  role: Role;
  email: string;
  logout: Function;
}

const AppLayout: React.FunctionComponent<IAppLayout> = ({ children, role, email, logout }) => {
  
  const [isNavOpen, setIsNavOpen] = React.useState(true);
  const [isMobileView, setIsMobileView] = React.useState(true);
  const [isNavOpenMobile, setIsNavOpenMobile] = React.useState(false);

  const onNavToggleMobile = () => {
    setIsNavOpenMobile(!isNavOpenMobile);
  };
  const onNavToggle = () => {
    setIsNavOpen(!isNavOpen);
  };
  const onPageResize = (props: { mobileView: boolean; windowSize: number }) => {
    setIsMobileView(props.mobileView);
  };
  const Header = (
    <PageHeader
      logo="Mass Open Cloud"
      headerTools={
        <PageHeaderTools>
          <a className={css(styles.pageHeaderBrandLink)}>
          {email || RoleMap[role]}
          </a>
          &nbsp;
          <Button onClick={() => logout()}> LOGOUT </Button>
        </PageHeaderTools>
      }
      showNavToggle
      isNavOpen={isNavOpen}
      onNavToggle={isMobileView ? onNavToggleMobile : onNavToggle}
    />
  );

  const Navigation = (
    <Nav id="nav-primary-simple" theme="dark">
      <NavList id="nav-list-simple">
        {routes.map(
          (route, idx) =>
            route.label && (
              <NavItem key={`${route.label}-${idx}`} id={`${route.label}-${idx}`}>
                <NavLink exact to={route.path} activeClassName="pf-m-current">
                  {route.label}
                </NavLink>
              </NavItem>
            )
        )}
      </NavList>
    </Nav>
  );
  const Sidebar = <PageSidebar theme="dark" nav={Navigation} isNavOpen={isMobileView ? isNavOpenMobile : isNavOpen} />;
  //const PageSkipToContent = <SkipToContent href="#primary-app-container">Skip to Content</SkipToContent>;

  return (
    <Page
      mainContainerId="primary-app-container"
      header={Header}
      sidebar={Sidebar}
      onPageResize={onPageResize}
     // skipToContent={PageSkipToContent}
    >
      {children}
    </Page>
  );
};

export { AppLayout };
