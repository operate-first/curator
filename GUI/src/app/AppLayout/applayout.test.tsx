import React from 'react';
import { shallow, mount } from 'enzyme';
import { AppLayout } from './AppLayout';
import { BrowserRouter as Router } from 'react-router-dom';
import { Role } from '..';

describe('AppLayout tests', () => {

  test('should render AppLayout component', () => {
    const layout = shallow(<AppLayout role={Role.ADMIN} logout={() => {}}><span /></AppLayout>);
    expect(layout).toMatchSnapshot();
  });


  test('should hide the sidebar when clicking the nav-toggle button', () => {
    const wrapper = mount(<Router><AppLayout role={Role.ADMIN} logout={() => {}}><span /></AppLayout></Router>);
    const button = wrapper.find('#nav-toggle').hostNodes();
    expect(wrapper.find('#page-sidebar').hasClass('pf-m-collapsed')).toBeTruthy();

    button.simulate('click');
    expect(wrapper.find('#page-sidebar').hasClass('pf-m-expanded')).toBeTruthy();
    expect(wrapper.find('#page-sidebar').hasClass('pf-m-collapsed')).toBeFalsy();
  });

  test('should auto-close nav in mobile view', () => {
    const wrapper = mount(<Router><AppLayout role={Role.ADMIN} logout={() => {}}><span /></AppLayout></Router>);
    global.innerWidth = 500;
    // Trigger the window resize event
    global.dispatchEvent(new Event('resize'));
    expect(wrapper.find('#page-sidebar').hasClass('pf-m-collapsed')).toBeTruthy();
  })

})
