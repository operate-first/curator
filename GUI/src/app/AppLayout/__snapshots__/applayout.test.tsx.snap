// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout tests should render AppLayout component 1`] = `
<Page
  defaultManagedSidebarIsOpen={true}
  header={
    <Unknown
      headerTools={
        <Unknown>
          <a
            className="pf-c-page__header-brand-link"
          >
            Admin
          </a>
          Â 
          <Unknown
            onClick={[Function]}
          >
             LOGOUT 
          </Unknown>
        </Unknown>
      }
      isNavOpen={true}
      logo="Mass Open Cloud"
      onNavToggle={[Function]}
      showNavToggle={true}
    />
  }
  isManagedSidebar={false}
  mainContainerId="primary-app-container"
  mainTabIndex={-1}
  onPageResize={[Function]}
  sidebar={
    <Unknown
      isNavOpen={false}
      nav={
        <Nav
          id="nav-primary-simple"
          onSelect={[Function]}
          onToggle={[Function]}
          theme="dark"
        >
          <NavList
            ariaLeftScroll="Scroll left"
            ariaRightScroll="Scroll right"
            id="nav-list-simple"
          >
            <Unknown
              id="Dashboard-0"
            >
              <NavLink
                activeClassName="pf-m-current"
                exact={true}
                to="/"
              >
                Dashboard
              </NavLink>
            </Unknown>
            <Unknown
              id="Openshift-1"
            >
              <NavLink
                activeClassName="pf-m-current"
                exact={true}
                to="/openshift"
              >
                Openshift
              </NavLink>
            </Unknown>
            <Unknown
              id="Support-2"
            >
              <NavLink
                activeClassName="pf-m-current"
                exact={true}
                to="/support"
              >
                Support
              </NavLink>
            </Unknown>
            <Unknown
              id="Reports-3"
            >
              <NavLink
                activeClassName="pf-m-current"
                exact={true}
                to="/reports"
              >
                Reports
              </NavLink>
            </Unknown>
          </NavList>
        </Nav>
      }
      theme="dark"
    />
  }
>
  <span />
</Page>
`;
